### PR TITLE
EN-2333 EN-2289 EN-2265 Moved code from core + DE API changes

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

**Comment by zonia3000** [comment]
This PR contains also the changes I did for the DE Test All endpoint (EN-2289 EN-2265). I had opened [a PR in core](https://github.com/entando/entando-core/pull/757), that I closed today. Maybe it is easier to have a look at the changes on that PR.

**Comment by zonia3000** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by rzsouza** [comment]
retest this please

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/client/DigitalExchangesClientImpl.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Override
+    public <R extends RestResponse<?, ?>, C> R getSingleResponse(DigitalExchange digitalExchange, DigitalExchangeCall<R, C> call) {
+        OAuth2RestTemplate restTemplate = digitalExchangesManager.getRestTemplate(digitalExchange.getName());
+        return new DigitalExchangeCallExecutor<>(messageSource, digitalExchange, restTemplate, call).getResponse();
+    }
+
+    @Override
+    public <R extends RestResponse<?, ?>, C> R getSingleResponse(String digitalExchangeName, DigitalExchangeCall<R, C> call) {
+        return getSingleResponse(digitalExchangesManager.findByName(digitalExchangeName).get(), call);
```
[the remark this answers is not in the document]
Comment (zonia3000) [reply plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/client/DigitalExchangesClientImpl.java]: I thrown an `IllegalArgumentException`. I think that when one calls this method he should know that a DE with that name should exists and be configured, so it should be a quite critical event if the Optional doesn't return the expected DE here.

File: plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/DigitalExchangeConstants.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system;
+
+public interface DigitalExchangeConstants {
```
[the remark this answers is not in the document]
Comment (zonia3000) [reply plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/DigitalExchangeConstants.java]: Old bad habits! ;-) Thanks

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by zonia3000** [comment]
retest this please

File: plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/client/DigitalExchangesClientImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rzsouza) [reply plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/client/DigitalExchangesClientImpl.java]: It makes sense

**Comment by entando-service** [comment]
All is Well

**Pull request merged**